src/main/java/mekanism/client/ClientTickHandler.java

```diff
@@ -241,6 +248,19 @@ public void tickStart() {
                 }
             }
 
+            if (isHydroEnhancementOn(minecraft.player)) {
+                hydroEnhancement = true;
+                // adds if it doesn't exist, otherwise tops off duration to 220. equal or less than 200 will make it flicker
+                minecraft.player.addEffect(new MobEffectInstance(MobEffects.DOLPHINS_GRACE, 220, 0, false, false, false));
```

Comment:
This can probably just be `200` like the night vision module used to be and remove the comment? The reason why night vision is at 220 is that 200 corresponds to when night vision starts flickering to warn that the potion is running out.

Reply:
I wasn't sure if it would flicker the effect if it was lower, but if it didn't I wasn't sure why not to go down to 1 tick? or maybe 2 depending on the order of clearing/updating effects.

Reply:
Mostly because vanilla is kind of strange/then it accounts for a bit of lag by giving it an extra second as it is permanent when active and instantly removed when turned off so it isn't like a couple ticks vs a second make a difference there.

Reply:
Would it be better to extract a constant like PERM_POTION_TICKS or something and keep them consistent?

Reply:
Hmm probably. I also feel like for dolphins grace though (even though client side controls movement) it should be added server side instead of client side (and let MC sync it to the client presuming it updates it properly when we reapply it). That way if some anticheat type mod is installed it is less likely to bug out.

Though actually thinking about it maybe instead of adding an effect like this we want to use forge's swim speed attribute (`ForgeMod.SWIM_SPEED.get()`)? Then it could even stack when the player is near a dolphin. (See `PlayerState#updateStepAssist` for roughly how it could probably be added/applied properly)

Reply:
I like that, I had initially discounted using it because it didn't affect the water push back like depth strider. But I like it much more than the potion effect

Reply:
I switched to the attribute, and it looks much nicer.

To compare to the speed of the locomotion unit I turned the attribute addition to 5 or 6x normal swim speed. So I looked into how that unit is working. As far as I can tell it is literally adding the boost value as meters per tick to the player's movement. Do we want to also change that to use the MOVEMENT_SPEED attribute while we're at it?

Reply:
Maybe, I definitely think it would be cleaner if it used the attribute though we will need to tweak it to make sure it properly still uses energy and also look at some of the interactions as I am not sure what places check the attribute. (Basically our can function/our boost throttling for when in the air or in water and then also the jump throttling in `CommonPlayerTickHandler` for if it is active and the hydraulic propulsion unit is also active)